### DISCUSSION TITLE
Draw fully customizable shapes

### DISCUSSION THREAD

**Discussion by schriker** [react-native-maps/react-native-maps#4643] [Q&A]
Hi 👋 

I'm playing around with this library trying to achieve something like this:

https://user-images.githubusercontent.com/28902144/226088752-b1ce4764-0d5a-496e-9097-56ecc406efc3.mp4

So basically user can draw fully customizable shapes (polygon let's say), with draggable and editable points. I've ended up using SVG and drawing them above the MapView, but the result is not 100% what I want, while the editing/dragging points is smooth enough, the animation when changing the map region is lagging (especially on android devices). 

Is something like the first video possible with this library? If yes, can someone point me in the right direction? Or I would be better off trying to do this on the fully native side. 

My results:

Android:

[android.webm](https://user-images.githubusercontent.com/28902144/226088624-84f64f59-9ca7-4820-9a4a-4fd55be8c644.webm)

iOS:

https://user-images.githubusercontent.com/28902144/226088639-565c45fb-91a6-4b83-8146-acec53904081.mp4

**Comment by sapcoiot00** [comment]
@schriker plz share this code

**Reply by cvramirez-nasa** [reply]
See post about [@siposdani87/expo-maps-polygon-editor](https://www.npmjs.com/package/@siposdani87/expo-maps-polygon-editor) below

**Comment by schriker** [comment]
@sapcoiot00 Sorry I do not have this code anymore this was more like personal challenge and I just removed it :| But basically what I was doing was using Reanimated, Gesture Handler and React Native SVG to draw custom polygon, and points (circles) above the MapView, then was using `pointForCoordinate` and `coordinateForPoint` from `https://github.com/react-native-maps/react-native-maps/blob/master/docs/mapview.md` to get correct coordinates based on points placement on device screen.

**Reply by cvramirez-nasa** [reply]
I have not tried this myself, but to anyone interested, here's a cool looking package,
[https://www.npmjs.com/package/@siposdani87/expo-maps-polygon-editor](https://www.npmjs.com/package/@siposdani87/expo-maps-polygon-editor)

I imagine some folks don't find it because they search for "react-native-maps-...", but this has "expo-maps-..." in the name.

![expo-maps-polygon-editor](https://github.com/react-native-maps/react-native-maps/assets/45186442/52628a9a-177b-4d17-aac6-8cc7f97890da)
